esp-hal/src/lib.rs

```diff
@@ -2,13 +2,55 @@
     docsrs,
     doc = "<div style='padding:30px;background:#810;color:#fff;text-align:center;'><p>You might want to <a href='https://docs.esp-rs.org/esp-hal/'>browse the <code>esp-hal</code> documentation on the esp-rs website</a> instead.</p><p>The documentation here on <a href='https://docs.rs'>docs.rs</a> is built for a single chip only (ESP32-C6, in particular), while on the esp-rs website you can select your exact chip from the list of supported devices. Available peripherals and their APIs change depending on the chip.</p></div>\n\n<br/>\n\n"
 )]
-//! Bare-metal (`no_std`) HAL for Espressif devices. Implements most of the
-//! traits defined by various packages in the [embedded-hal] repository.
+//! Bare-metal (`no_std`) HAL for all Espressif ESP32 devices.
+//! The HAL implements both blocking _and_ async APIs for many
+//! peripherals. Where applicable, driver implement the [embedded-hal]
+//! and [embedded-hal-async] traits.
 //!
+//! This documentation is built for the
+#![cfg_attr(esp32, doc = "**esp32**")]
+#![cfg_attr(esp32s3, doc = "**esp32s3**")]
+#![cfg_attr(esp32s2, doc = "**esp32s2**")]
+#![cfg_attr(esp32c2, doc = "**esp32c2**")]
+#![cfg_attr(esp32c3, doc = "**esp32c3**")]
+#![cfg_attr(esp32c6, doc = "**esp32c6**")]
+#![cfg_attr(esp32h2, doc = "**esp32h2**")]
+//! please ensure you are reading the correct [documentation] for your target
+//! device.
+//!
+//! ## Choosing a device
+//!
+//! Depending on your target device, you need to enable the chip feature
+//! for that device. You may also need to do this on other ancillary esp crates.
+//!
+//! ## Examples
+//!
+//! We have a plethora of [examples] in the esp-hal repository. We use
+//! an [xtask] automate the building, running and testing of code and examples
+//! within esp-hal.
+//!
+//! Invoke the following command in the root of the esp-hal repository to get
```

Comment:
Since the chapter is about the examples, we could also show the full command to run an example?

Most of the other commands are probably not too useful for people who just want to use the HAL. Maybe `build-documentation` - just an idea

Reply:
I was thinking about that but also doing it this way means we don't have to keep updating it if the xtask changes. I'm happy to make the change, but I'll see what everyone else things and go with the majority.

Reply:
If we don't agree on anything now, we can just keep it as is - we can change that later. It's definitely not wrong